tools/packaging/static-build/qemu-virtiofs/Dockerfile

```diff
@@ -49,23 +49,17 @@ RUN cd  .. && git clone "${QEMU_VIRTIOFS_REPO}" qemu-virtiofs
 RUN git checkout "${QEMU_VIRTIOFS_TAG}"
 
 ADD scripts/configure-hypervisor.sh /root/configure-hypervisor.sh
-ADD qemu/patches/ /root/kata_qemu_patches
+ADD qemu /root/kata_qemu
 
 # Apply experimental specific patches
 # Patches to quick fix virtiofs fork
-ENV VIRTIOFS_PATCHES_DIR=/root/kata_qemu_patches/${QEMU_VIRTIOFS_TAG}/
-RUN find "${VIRTIOFS_PATCHES_DIR}" -name '*.patch' -type f |sort -t- -k1,1n > patches_virtiofs
-RUN echo "Patches to apply for virtiofs fixes:"
-RUN cat patches_virtiofs
-RUN [ ! -s patches_virtiofs ] || git apply $(cat patches_virtiofs)
-
-RUN cat VERSION | awk 'BEGIN{FS=OFS="."}{print $1 "." $2 ".x"}' > stable_branch
-RUN echo "root/kata_qemu_patches/$(cat stable_branch)/" > patches_qemu_dir
-RUN echo "patches dir $(cat patches_qemu_dir)"
-RUN find "$(cat patches_qemu_dir)" -name '*.patch' -type f |sort -t- -k1,1n > patches_qemu
-RUN echo "Patches to apply for qemu:"
-RUN cat patches_qemu
-RUN [ ! -s patches_qemu ] || git apply $(cat patches_qemu )
+ENV VIRTIOFS_PATCHES_DIR=/root/kata_qemu/patches/${QEMU_VIRTIOFS_TAG}/
```

Comment:
what is `QEMU_VIRTIOFS_TAG` suppose to be set to? It looks like we might end up applying the same patch twice if ` QEMU_VIRTIOFS_TAG` is set to the same as qemu `stable_branch`?

Reply:
@bergwolf according to 'versions.yaml` the value of `QEMU_VIRTIOFS_TAG` will be `qemu5.0-virtiofs-with51bits-dax`.

It seems unlikely to have same value of the `stable_branch`. The stable branch will have a value like `5.1.x` and unless it has a branch named `5.1.x` (which is ugly) then the values won't clash. Well, if you want I can put a guard...better safe than sorry :)

Reply:
But we do not have these patches in tree, right? Are these added to apply some external patches? It's a bit confusing though that the version is maintained in tree but no actual patches are committed.

Looks like we should be able to just clone and use `qemu5.0-virtiofs-with51bits-dax` tagged code from `https://gitlab.com/virtio-fs/qemu.git`. Then `VIRTIOFS_PATCHES_DIR` is something obsolete and can be removed?

Reply:
This  mechanism to apply virtio-fs specific patches was introduced in commit 04b156f60.

Currently we don't have any specific patch, i.e, a checkout from `qemu5.0-virtiofs-with51bits-dax` branch just works. But suppose that into the future there should be applied patches on `qemu5.0-virtiofs-with51bits-dax` branch which aren't merged into `https://gitlab.com/virtio-fs/qemu.git` first, then we just drop the patches in the `qemu/patches/qemu5.0-virtiofs-with51bits-dax` which they will be applied (no changes to Dockerfile needed).

In this PR I am just keeping the same logic but I can change it. IMO this mechanism can be kept.

Reply:
Fair enough! Thanks for the explanation.